deafrica/monitoring/s2_gap_filler.py

```diff
@@ -82,7 +109,152 @@ def get_common_message_attributes(stac_doc: Dict) -> Dict:
     return msg_attributes
 
 
-def prepare_message(scene_paths: list, log: Optional[logging.Logger] = None):
+def prepare_s2_l2a_stac(src_stac_doc: Dict):
+    """Prepares the appopriate STAC data to send with the sqs message.
+    The original json/stac_doc must be modified with the appropriate
+    fields. The metadata changes across time which makes this a bit
+    messy for capturing known edge cases.
+
+    Args:
+        src_stac_doc: Original STAC doc/json from gap report
+
+    Returns:
+        stac_metadata: formatted stacmetadata for the sns message
+    """
+
+    # change the properties to align with original sqs message
+    # read in the tileinfo.json file and create a STAC document
+    # using the sentinel_s2_l2a package.
+    # This will form the basis of our SNS message body as it is
+    # most closely aligned with the original message
+    if "tileinfo_metadata" in list(src_stac_doc["assets"].keys()):
+        tileinfo_url = src_stac_doc["assets"]["tileinfo_metadata"]["href"]
+    else:
+        tileinfo_url = src_stac_doc["assets"]["info"]["href"]
+
+    tileinfo = requests.get(tileinfo_url, stream=True)
+    tileinfo = json.loads(tileinfo.text)
+
+    # update the base url to generate STAC metadata for source
+    base_url = f"https://roda.sentinel-hub.com/sentinel-s2-l2a/{tileinfo['path']}"
+    tileinfo_metadata = sentinel_s2_l2a(tileinfo, base_url=base_url)
+
+    # add the links from the tileinfo
+    relevant_links = ["self", "canonical", "derived_from"]
+    links = [x for x in src_stac_doc["links"] if x["rel"] in relevant_links]
+
+    # update the link path to the origin
+    SENTINEL_2_COGS_URL = "https://sentinel-cogs.s3.us-west-2.amazonaws.com"
+    for link in links:
+        link["href"] = link["href"].replace("s3://sentinel-cogs", SENTINEL_2_COGS_URL)
```

Comment:
comparing between
1) upstream source
https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/33/M/TT/2019/3/S2A_33MTT_20190327_0_L2A/S2A_33MTT_20190327_0_L2A.json

2) deafrica
https://deafrica-sentinel-2.s3.af-south-1.amazonaws.com/sentinel-s2-l2a-cogs/33/M/TT/2019/3/S2A_33MTT_20190327_0_L2A/S2A_33MTT_20190327_0_L2A.json


it looks like it doesn't get changed from `s3://` to `https://`? 

and should it be replaced with `deafrica-sentinel-2`?

Reply:
Yes but this change gets made in the lambda when the files are copied across from source to our bucket: 

https://github.com/digitalearthafrica/datakube-serverless/blob/main/deafrica-sync/sync/s2.py#L183 

We therefore need to ensure the links that we provide in the SQS messages from thr s1_gap_filler.py point to the source cogs as http (`SENTINEL_2_COGS_URL`). These links are then updated and replaced with `deafrica-sentinel-2` in the lambda.

I hope this is what you are referring to.

Reply:
Ah gotcha, I forgot this gap_filler writes into the deafrica-sync process